test/test_ops.py

```diff
@@ -1577,6 +1577,121 @@ def is_bit_set(x):
+
+                requires_grad = (
+                    op.supports_autograd
+                    and dtype in op.supported_backward_dtypes(torch.device(device).type)
+                    # TODO: OpInfo really ought to error out for this case, but it's
+                    # not exercised in test_ops_gradients atm.  The problem is not
+                    # complex32 per-se (which is supported by data movement only ops)
+                    # but that when we do backwards we expect other ops like add to work
+                    and not dtype == torch.complex32
+                )
+
+                samples = op.sample_inputs(device, dtype, requires_grad=requires_grad)
+                for sample_input in samples:
+                    args = [sample_input.input] + list(sample_input.args)
+                    kwargs = sample_input.kwargs
+
+                    try:
+                        with CollectDispatchMode():
+                            out = func(*args, **kwargs)
+                            out.sum().backward()
+                    except Exception as e:
+                        pass
+
+                    if inplace_func:
+                        try:
+                            with CollectDispatchMode():
+                                out = inplace_func(*args, **kwargs)
+                                out.sum().backward()
+                        except Exception as e:
+                            pass
+
+        remaining_aten_ops = (all_aten_ops - seen_aten_ops)
+        untested_aten_ops = set()
+
+        for aten_op in remaining_aten_ops:
+            overload = aten_op.overloads()[0]
+            overload = getattr(aten_op, overload)
+
+            try:
+                if not torch._ops.has_key(overload, "CompositeImplicitAutograd"):  # noqa: W601
+                    untested_aten_ops.add(aten_op)
+                    continue
+            except Exception as e:
+                untested_aten_ops.add(aten_op)
+
+        known_untested_aten_ops = {
+            aten._embedding_bag_forward_only,
+            aten._fused_dropout,
+            aten._fused_moving_avg_obs_fq_helper,
+            aten._list_to_tensor,
+            aten._nested_tensor_from_tensor_list,
+            aten._pin_memory,
+            aten._reshape_alias,
+            aten._resize_output_,
+            aten._resize_output,
+            aten._sparse_coo_tensor_with_dims_and_tensors,
+            aten.cudnn_batch_norm,
+            aten.cudnn_batch_norm_backward,
+            aten.empty_strided,
+            aten.fill,
+            aten.hardswish_,
+            aten.hardtanh_,
+            aten.is_pinned,
+            aten.is_same_size,
+            aten.lift,
+            aten.quantized_gru,
+            aten.quantized_lstm,
+            aten.rand_like,
+            aten.randperm,
+            aten.relu_,
+            aten.uniform,
+        }
+
+        unexpected_untested_aten_ops = untested_aten_ops - known_untested_aten_ops
```

Comment:
Shouldn't this test the `symmetric_difference` to keep the handcoded list up-to-date? Or it could test both differences to produce nicer error messages so developers know what to do